core/app/models/spree/store_credit.rb

```diff
@@ -30,6 +31,7 @@ class Spree::StoreCredit < Spree::Base
   scope :order_by_priority, -> { includes(:credit_type).order('spree_store_credit_types.priority ASC') }
 
   after_save :store_event
+  after_save :create_ledger_entry
```

Comment:
I'm not super keen on building everything around callbacks. Could we try extracting the logic to some specific methods so I could do the following:

```Ruby
store_credit.credit(100, admin_user_who_granted_this)
store_credit.debit(50, payment)
```

Then those methods could create the ledger entry and modify the current balance.